src/reporting/branch.se

```diff
@@ -139,3 +139,13 @@ def isContainerForReportingToken(shadyReportingToken: address):
 def isParentOf(shadyChild: address):
     branchId = shadyChild.getParentPayoutDistributionHash()
     return(self.childBranches[branchId] == shadyChild)
+
+def isContainerForShareToken(shadyShareToken):
+    shadyMarket = shadyShareToken.getMarket()
+    if not shadyMarket:
+        return(0)
+    if not self.isContainerForMarket(shadyMarket):
+        return(0)
+    if not shadyMarket.isContainerForShareToken(shadyToken):
```

Comment:
For clarity, there may be value in assigning to a local variable named `legitMarket`.  This is what I believe is done in the others, and it helps the reader understand that "at this point in the code, we now trust this market".
```
if not self.isContainerForMarket(shadyMarket):
    return(0)
legitMarket = shadyMarket
if not legitMarket.isContainerForShareToken(shadyToken):
```

Perhaps a better name would be `trustedMarket`... I kind of wish I had used that in similar functions.